File: javascript/high-scores/high-scores.js
//High scores exercise
export class HighScores {
  constructor(input) {
    this.input = input 
  }

  get scores() {
    return this.input
  }

  get latest() {
    return (this.input)
      .sort((a, b) => a-b)
      .filter((i, j) => (i != 0))[0]
  }

  get personalBest() {
    return (this.input)
      .sort((a, b) => b-a)[0]
  }

  get personalTopThree() {
    return (this.input)
      .sort((a, b) => b-a)
      .filter((i, j) => (j < 3))
  }
}

File: ruby/high-scores/high_scores.rb
# frozen_string_literal: true

# HighScore exercism
class HighScores
  attr_reader :scores

  def initialize(scores)
    @scores = scores.delete_if(&:zero?)
  end

  def latest
    scores.min
  end

  def personal_best
    scores.sort.max
  end

  def personal_top_three
    scores.sort.reverse.take(3)
  end

  def latest_is_personal_best?
    scores.last == scores.max
  end
end
